### PR TITLE
TSPS-65 Use JPA best practices for table structure and use test containers in tests

### PR DESCRIPTION
**Pull request by jsotobroad**
There are a couple of things going on in this PR.

Removed any varchar columns are they are just not preferred in the context of postgres 13 and converted them to text - https://maximorlov.com/char-varchar-text-postgresql/

instead of adding migrations i just edited the ones that were existing and removed the varchar migration.  this will require us to clear the dev database before merging

Jobs table
- added id column as primary auto incrementing key
- changed job_id to UUID type since it is a UUID
- added unique constraint to job_id column

Pipelines table
- added id column as primary auto incrementing key
- added version column
- added composite unique key contraint on pipieline_id + version

Updated tests to use postgres test container, this means in order to run tests you need to have docker available (this shouldnt be an issue)

### CODE REVIEW

File: service/src/test/resources/application-test.yml [hunk 1]
```diff
@@ -16,16 +7,17 @@ terra.common:
 # specifying the in-memory db for the "test" profile
```
Comment (jsotobroad) [inline service/src/test/resources/application-test.yml:7]: correct comment

**Comment by jsotobroad** [comment]
turns out removing that JDBC dependency made it so we didnt need those options anymore in the test application yaml so its been removed!

**Review by pshapiro4broad — COMMENTED** [review]
Already merged but I had a couple comments of things you may want to look at

File: service/src/main/java/bio/terra/pipelines/app/controller/JobsApiController.java [hunk 2]
```diff
@@ -94,7 +94,7 @@ public ResponseEntity<ApiPostJobResponse> createJob(
 
   @Override
   public ResponseEntity<ApiGetJobResponse> getJob(
-      @PathVariable("pipelineId") String pipelineId, @PathVariable("jobId") String jobId) {
+      @PathVariable("pipelineId") String pipelineId, @PathVariable("jobId") UUID jobId) {
```
[the remark this answers is not in the document]
Comment (jsotobroad) [reply service/src/main/java/bio/terra/pipelines/app/controller/JobsApiController.java:97]: are you referring to `@PathVariable` ?  is the concern here duplicate code for no reason or something more nefarious (assuming you were referring to `@PathVariable`

File: service/src/main/java/bio/terra/pipelines/service/JobsService.java [hunk 3]
```diff
@@ -88,15 +89,18 @@ protected UUID writeJobToDb(
...
   protected UUID writeJobToDbRetryDuplicateException(Job job) {
-    Optional<Job> jobExists = jobsRepository.findJobByJobId(job.getJobId());
-    if (jobExists.isPresent()) {
-      logger.warn("Duplicate jobId {} found, retrying", job.getJobId());
-      return null;
+    try {
+      jobsRepository.save(job);
+      logger.info("job saved for jobId: {}", job.getJobId());
+    } catch (DataIntegrityViolationException e) {
+      if (e.getCause().getClass().equals(ConstraintViolationException.class)) {
```
[the remark this answers is not in the document]
Comment (jsotobroad) [reply service/src/main/java/bio/terra/pipelines/service/JobsService.java:96]: good point, nothing seems to subclasses `ConstraintViolationException` now but might in the future. adding it to https://github.com/DataBiosphere/terra-scientific-pipelines-service/pull/35

[the remark this answers is not in the document]
Comment (jsotobroad) [reply service/src/main/java/bio/terra/pipelines/service/JobsService.java:96]: hmm thinking it about it a little more i could see a world where i wouldnt want to automatically have any subclasses of this follow this path of the code depending on how someone designed that subclassing but this is pretty much an academic scenario im cool with updating it to `instanceof`

**Pull request merged**